test/util.cpp

```diff
@@ -0,0 +1,79 @@
+//
```

Comment:
I followed suit of naming the test file after the header file, but `util` is a bit generic for a test suite. Do you prefer this or to put into a file called `comp`, for example? (Maybe makes sense to lift the production code out of `util` now it's getting a bit bigger...)